packages/network-controller/src/NetworkController.test.ts

```diff
@@ -1,224 +1,3653 @@
-import * as sinon from 'sinon';
-import Web3ProviderEngine from 'web3-provider-engine';
+import assert from 'assert';
+import { mocked } from 'ts-jest/utils';
 import { ControllerMessenger } from '@metamask/base-controller';
-import { NetworkType, NetworksChainId } from '@metamask/controller-utils';
+import * as ethQueryModule from 'eth-query';
+import Subprovider from 'web3-provider-engine/subproviders/provider';
+import createInfuraProvider from 'eth-json-rpc-infura/src/createProvider';
+import type { ProviderEngine } from 'web3-provider-engine';
+import createMetamaskProvider from 'web3-provider-engine/zero';
+import { Patch } from 'immer';
+import {
+  FakeProviderEngine,
+  FakeProviderStub,
+} from '../tests/fake-provider-engine';
 import {
   NetworkController,
-  NetworkControllerMessenger,
+  NetworkControllerActions,
+  NetworkControllerEvents,
   NetworkControllerOptions,
+  NetworkState,
   ProviderConfig,
 } from './NetworkController';
 
-const RPC_TARGET = 'http://foo';
-
-const setupController = (
-  pType: NetworkType,
-  messenger: NetworkControllerMessenger,
-) => {
-  const networkControllerOpts: NetworkControllerOptions = {
-    infuraProjectId: 'foo',
-    state: {
-      network: '0',
-      providerConfig: {
-        type: pType,
-        chainId: NetworksChainId[pType],
-      },
-      properties: { isEIP1559Compatible: false },
-    },
-    messenger,
+jest.mock('eth-query', () => {
+  return {
+    __esModule: true,
+    default: jest.requireActual('eth-query'),
   };
-  const controller = new NetworkController(networkControllerOpts);
-  controller.providerConfig = {} as ProviderConfig;
-  return controller;
-};
+});
+jest.mock('web3-provider-engine/subproviders/provider');
+jest.mock('eth-json-rpc-infura/src/createProvider');
+jest.mock('web3-provider-engine/zero');
 
-describe('NetworkController', () => {
-  let messenger: NetworkControllerMessenger;
+type WithControllerCallback<ReturnValue> = ({
```

Comment:
Nit: Moving these types closer to the function where they're used would make them easier to understand.

Reply:
Sure! Moved in [e0a4b38](https://github.com/MetaMask/core/pull/1039/commits/e0a4b38ccadd41d32df688434824659c8ff50b42).